include/seqan/bam_io/read_bam.h

```diff
@@ -97,6 +97,7 @@ readHeader(BamHeader & header,
            Bam const & /*tag*/)
 {
     clear(header);
+    FileExtensions<BgzfFile>::VALUE[0] = ".bam";
     // Read BAM magic string.
```

Comment:
Do you actually edit the static variable here? This is a hack at best. It will break other things, e.g. you won't be able to read a .bgzf file after that. It would also produce all sorts of problems in a multi-threaded context.

Please find a different solution :pray: 

Thanks!